46-sugar-ant/technical_specification.md

```diff
@@ -0,0 +1,300 @@
+The claims repository also has endpoints at
+- `GET /download-access/users/{user_id}/`
+
+The routes of these endpoints do not need to be changed. However, these endpoints must now also check that the corresponding claims are bound to IVAs that have been verified for the given user.
+
+### Access Request Service
+
+The `AccessRequest` model should have a new optional `iva_id` field, as specified in the section on backend model changes below.
+
+The `PATCH /access-requests/{access_request_id}` endpoint that is used to allow or deny access requests must be extended so that it also accepts the corresponding `iva_id` in the body. The `iva_id` must then also be passed to the claims repository.
+
+### Internal Auth Token
+
+Until now, an internal auth token was created and passed by the auth service after the user was successfully authenticated via LS Login. The token contained an additional `status` enum (active, inactive, invalid). It could contain the internal or external user id in the fields `id` and `ext_id`.
+
+In the new implementation, we change the internal auth token as follows:
+
+The internal auth token will be only created and passed on by the Auth Adapter if the use is fully authenticated, i.e. logged in via LS Login and the second factor has been validated.
+
+The `status` field will be removed from the auth token. The existence of the token always implies that the user account is active and not invalid.
+
+The `ext_id` field will be removed from the auth token. There are only two exceptional cases where it is needed, and in theses cases it can be stored in the `id` field instead. The `id` field will also be made mandatory and required to be a non-empty string.
+
+There are only the following three exceptions where the auth token will be also added by the Auth Adapter if the user is only logged in via LS Login:
+
+- `POST /users`
+	- used to self-register a user
+	- the `id` field of the auth context will contain the external id, not the internal id
+- `PUT /users/{user_id}`
+	- when requested by users to confirm a name an email change
+	- the user must be already registered
+	- the `id` field of the auth context will contain the internal id of the user, and it must correspond to the `user_id` in the path
+- `GET /users/{ext_id}`
+	- when requested by users to check their own registered data
+	- the `id` field of the auth context will contain the external id, not the internal id, and it must correspond to the `ext_id` in the path
+
+### Service Commons Library
+
+The `ghga-service-commons` library must be changed to reflect the changes in the internal auth token as outlined above.
+
+### Backend Models
+
+A new `IVA` (independent verification address) model must be added to the User Management service:
+	- `id`: string (unique internal id)
+	- `user_id`: string (internal id of the user)
+	- `type`: enum (phone, fax, postal address, in-person)
+	- `value`: string (the actual phone number)
+	- `verification_code`:  string (random number)
```

Comment:
As hash

Reply:
Changed to only store the hash of the verification code. This made adding another state necessary though.